commit 3dcafdd65b68f097cc4262905f8460842ef05b30
Author: Ryan Burns <rtburns@jpl.nasa.gov>
Date:   Wed Jun 30 11:15:56 2021 -0700

    Fix compilation error due to API change in HDF5 v1.12
    
    HDF5 v1.12 changed the signature of H5Ovisit_by_name.
    Using the new API is actually more efficient, since we can avoid
    retrieving fields that aren't needed for the visitor function.
    
    I'm also checking H5_USE_110_API so that anyone explicitly opting
    for the old API won't be affected by this change.

diff --git a/cxx/isce3/io/IH5.cpp b/cxx/isce3/io/IH5.cpp
index 4c8782c1..51a7ec98 100644
--- a/cxx/isce3/io/IH5.cpp
+++ b/cxx/isce3/io/IH5.cpp
@@ -108,8 +108,15 @@ std::vector<std::string> findByName(hid_t loc_id, std::string nameIn,
             findMetaInfo.basePath.append("/");
     }
 
+// If we are using the new H5Ovisit_by_name API introduced in HDF5 1.12:
+#if H5_VERSION_GE(1, 12, 0) && !defined(H5_USE_110_API)
+    // we do a more efficient visit, only retrieving the basic info
+    H5Ovisit_by_name3(loc_id, start.c_str(), H5_INDEX_NAME, H5_ITER_INC,
+                      matchName, &findMetaInfo, H5O_INFO_BASIC, H5P_DEFAULT);
+#else
     H5Ovisit_by_name(loc_id, start.c_str(), H5_INDEX_NAME, H5_ITER_INC,
                      matchName, &findMetaInfo, H5P_DEFAULT);
+#endif
 
     return findMetaInfo.outList;
 }
